featurebyte/service/feature_list_facade.py

```diff
@@ -0,0 +1,241 @@
+from featurebyte.service.feature_list import FeatureListService
+from featurebyte.service.feature_list_namespace import FeatureListNamespaceService
+from featurebyte.service.feature_list_status import FeatureListStatusService
+from featurebyte.service.feature_readiness import FeatureReadinessService
+from featurebyte.service.version import VersionService
+
+
+class FeatureListFacadeService:
+    """
+    Feature List Facade Service which is responsible for handling high level feature list operations,
+    and delegating the actual operations to the corresponding services.
+    """
+
+    def __init__(
+        self,
+        feature_list_service: FeatureListService,
+        feature_list_namespace_service: FeatureListNamespaceService,
+        feature_list_status_service: FeatureListStatusService,
+        feature_readiness_service: FeatureReadinessService,
+        version_service: VersionService,
+        default_version_mode_service: DefaultVersionModeService,
+    ):
+        self.feature_list_service = feature_list_service
+        self.feature_list_namespace_service = feature_list_namespace_service
+        self.feature_list_status_service = feature_list_status_service
+        self.feature_readiness_service = feature_readiness_service
+        self.version_service = version_service
+        self.default_version_mode_service = default_version_mode_service
+
+    async def create_feature_list(self, data: FeatureListServiceCreate) -> FeatureListModel:
+        """
+        Create a new feature list
+
+        Parameters
+        ----------
+        data: FeatureListServiceCreate
+            Feature list service create payload
+
+        Returns
+        -------
+        FeatureListModel
+        """
+        document = await self.feature_list_service.create_document(data=data)
+        await self.feature_readiness_service.update_feature_list_namespace(
+            feature_list_namespace_id=document.feature_list_namespace_id,
+            return_document=False,
+        )
+        output = await self.feature_list_service.get_document(document_id=document.id)
+        return output
+
+    async def create_new_version(self, data: FeatureListNewVersionCreate) -> FeatureListModel:
+        """
+        Create a new version of a feature list
+
+        Parameters
+        ----------
+        data: FeatureListNewVersionCreate
+            Feature list new version create payload
+
+        Returns
+        -------
+        FeatureListModel
+        """
+        document = await self.version_service.create_new_feature_list_version(data=data)
+        await self.feature_readiness_service.update_feature_list_namespace(
+            feature_list_namespace_id=document.feature_list_namespace_id,
+            return_document=False,
+        )
+        output = await self.feature_list_service.get_document(document_id=document.id)
+        return output
+
+    async def make_feature_production_ready(
```

Comment:
nit: maybe more accurate to be `make_features_production_ready` since this applies to all features in the feature list?